### DISCUSSION TITLE
Slack variables value and multipliers

### DISCUSSION THREAD

**Discussion by juanmanzanero** [coin-or/Ipopt#574] [Q&A] [answered]
Hello!

I am trying to do some sensitivity analysis to optimal solutions I get from Ipopt, but I am struggling to get the slack variables and their associated multipliers. 

In the finalize_solution method, I save them as:

```
// Return slack variables
const Ipopt::Number* s_values = dynamic_cast<const Ipopt::DenseVector*>(GetRawPtr(ip_data->curr()->s()))->Values();
solution_.s = Dvector(s_values, s_values + ip_data->curr()->s()->Dim());

const Ipopt::Number* vl_values = dynamic_cast<const Ipopt::DenseVector*>(GetRawPtr(ip_data->curr()->v_L()))->Values();
solution_.vl = Dvector(vl_values, vl_values + ip_data->curr()->v_L()->Dim());

const Ipopt::Number* vu_values = dynamic_cast<const Ipopt::DenseVector*>(GetRawPtr(ip_data->curr()->v_U()))->Values();
solution_.vu = Dvector(vu_values, vu_values + ip_data->curr()->v_U()->Dim());
```

But I find that the slack variables I get they are outside the range given by c_lb < s < c_ub (in my case, if c_ub = 2.0, the slack variable is 2.000000017493693)

The problem with this, is that the Hessian matrix I will compute for my post analysis is not definite positive, since the term v_u / (c_ub - s) < 0 (being v_u the associated upper bound multiplier).

So my first question is then: It is expected that the slack variables get out of bounds?

Many thanks in advance!!!

**Comment by juanmanzanero** [comment, marked as answer]
I just saw that setting `relax_bound_factor` to 0 solves the issue. Just for my education, what side effect can have setting this value to zero?

Cheers,

Juan.

**Reply by svigerske** [reply]
It will prevent the constraint to be satisfied with equality, since such points won't be "interior" anymore.
